### ISSUE TITLE
Unable to install on OpenSUSE MicroOS

### ISSUE THREAD

**Issue by sith-on-mars** [mullvad/mullvadvpn-app#4636]
## Issue report

Operating system: OpenSUSE MicroOS
App version: 2023.3

### Issue description

I couldn't install Mullvad on MicroOS. It complained that `/opt/Mullvad VPN/resources/mullvad-setup` and `mullvad-daemon.service` didn't exist. On MicroOS the package is installed only after reboot. I'm wondering if you would be able to fix it or if there is any workaround?

```
dadstick@localhost:~> sudo transactional-update pkg in '/home/dadstick/Downloads/MullvadVPN-2023.3_x86_64.rpm' 
[sudo] password for root: 
Checking for newer version.
transactional-update 4.1.5 started
Options: pkg in /home/dadstick/Downloads/MullvadVPN-2023.3_x86_64.rpm
Separate /var detected.
2023-04-30 02:40:25 tukit 4.1.5 started
2023-04-30 02:40:25 Options: -c6 open 
2023-04-30 02:40:25 Using snapshot 6 as base for new snapshot 7.
2023-04-30 02:40:25 Syncing /etc of previous snapshot 5 as base into new snapshot "/.snapshots/7/snapshot"
2023-04-30 02:40:25 SELinux is enabled.
ID: 7
2023-04-30 02:40:27 Transaction completed.
Calling zypper install
2023-04-30 02:40:29 tukit 4.1.5 started
2023-04-30 02:40:29 Options: callext 7 zypper -R {} install /home/dadstick/Downloads/MullvadVPN-2023.3_x86_64.rpm 
2023-04-30 02:40:31 Executing `zypper -R /tmp/transactional-update-tityQW install /home/dadstick/Downloads/MullvadVPN-2023.3_x86_64.rpm`:
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: nothing provides 'dbus-libs' needed by the to be installed mullvad-vpn-2023.3-1.x86_64
 Solution 1: do not install mullvad-vpn-2023.3-1.x86_64
 Solution 2: break mullvad-vpn-2023.3-1.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/c/d/?] (c): 2
Resolving dependencies...
Resolving package dependencies...

The following 2 NEW packages are going to be installed:
  libnotify-tools mullvad-vpn

2 new packages to install.
Overall download size: 74.3 MiB. Already cached: 0 B. After the operation, additional 297.6 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y

Checking for file conflicts: [...done]
Warning: 2 packages had to be excluded from file conflicts check because they are not yet downloaded.

    Note: Checking for file conflicts requires not installed packages to be downloaded in advance in
    order to access their file lists. See option '--download-in-advance / --dry-run --download-only'
    in the zypper manual page for details.

Retrieving: libnotify-tools-0.8.2-1.2.x86_64 (openSUSE-Tumbleweed-Oss) (1/2),  20.0 KiB    
Retrieving: libnotify-tools-0.8.2-1.2.x86_64.rpm [...done]
(1/2) Installing: libnotify-tools-0.8.2-1.2.x86_64 [..done]
Retrieving: mullvad-vpn-2023.3-1.x86_64 (Plain RPM files cache) (2/2),  74.3 MiB    
MullvadVPN-2023.3_x86_64.rpm:
    Package header is not signed!

mullvad-vpn-2023.3-1.x86_64 (Plain RPM files cache): Signature verification failed [6-File is unsigned]
Abort, retry, ignore? [a/r/i] (a): i
(2/2) Installing: mullvad-vpn-2023.3-1.x86_64 [..
/var/tmp/rpm-tmp.2lrKnm: line 6: /opt/Mullvad VPN/resources/mullvad-setup: No such file or directory
Running in chroot, ignoring command 'stop'
Failed to disable unit, unit mullvad-daemon.service does not exist.
error: %prein(mullvad-vpn-2023.3-1.x86_64) scriptlet failed, exit status 1
error: mullvad-vpn-2023.3-1.x86_64: install failed
error]
Installation of mullvad-vpn-2023.3-1.x86_64 failed:
Error: Subprocess failed. Error: RPM failed: Command exited with status 1.
Abort, retry, ignore? [a/r/i] (a): i
Executing %posttrans script 'MullvadVPN-2023.3_x86_64.rpm' [..
not-found
Failed to enable unit, unit /usr/lib/systemd/system/mullvad-daemon.service does not exist.
Running in chroot, ignoring command 'start'
Failed to enable unit, unit /usr/lib/systemd/system/mullvad-early-boot-blocking.service does not exist.
.done]
2023-04-30 02:41:06 Application returned with exit status 107.
ERROR: zypper install on /.snapshots/7/snapshot failed with exit code 107!
Removing snapshot #7...
2023-04-30 02:41:06 tukit 4.1.5 started
2023-04-30 02:41:06 Options: abort 7 
2023-04-30 02:41:08 Discarding snapshot 7.
2023-04-30 02:41:08 Transaction completed.
transactional-update finished
```

**Comment by faern** [comment]
Sorry, but we don't support OpenSUSE, nor MicroOS. Our installation scripts rely on the needed resources being in place while executing. Why is MicroOS running the post installation scripts before finishing the install? That sounds backwards.

**Comment by firefoxlover** [comment]
It works normally on Fedora Atomic, which is strange. But there the postinstall script doesnt run at all.

**Comment by MarkusPettersson98** [comment]
Hello again. As mentioned, we don't officially support OpenSUSE, and we currently don't have any plans on doing so either. I'll go ahead and close this issue, but feel free to open a new issue for adding support for OpenSUSE if you would like to see that. It will be easier for us to track the engagement from the community in a more focused issue. Thanks! 💛

**Comment by bernado82** [comment]
On openSUSE **Tumbleweed**, Mullvad VPN 2025.x can be installed directly _"out-of-the-box"_ from Fedora RPM package (but **NOT on Leap 15.6**, rebuilt still required!). I've **not tested** on openSUSE MicroOS!

If you want install MullvadVPN from original Fedora RPM, you can import official **GPG signing key** with the command:

`sudo rpm --import https://mullvad.net/media/mullvad-code-signing.asc`

or `wget https://mullvad.net/media/mullvad-code-signing.asc` (and by importing manually the GPG key file **mullvad-code-signing.asc** with **YaST repositories manager**)

If you want to get **automatic updates** with **Zypper** and **Discover**, you've just to add official Fedora repository with the command:

`sudo zypper ar -f https://repository.mullvad.net/rpm/stable/x86_64/ MullvadVPN`

and this one if you want install beta versions also:

`sudo zypper ar -f https://repository.mullvad.net/rpm/beta/x86_64/ MullvadVPN_beta`

![Image](https://github.com/user-attachments/assets/2484063b-f620-4541-9326-2e2980b01316)
